apps/api/src/app/workflows-v2/usecases/preview/preview.usecase.ts

```diff
@@ -61,13 +59,7 @@ export class PreviewUsecase {
 
       payloadExample = this.payloadProcessor.enhanceEventCountValue(payloadExample);
 
-      // Add resolved context to payload example for schema building
-      if (context.resolvedContext && Object.keys(context.resolvedContext).length > 0 && !payloadExample.context) {
-        payloadExample.context = context.resolvedContext;
-      }
-
       const cleanedPayloadExample = this.payloadProcessor.cleanPreviewExamplePayload(payloadExample);
-      const schema = await this.schemaBuilder.buildPreviewPayloadSchema(payloadExample, context.workflow.payloadSchema);
```

Comment:
This is the main simplification point 

We had two places where very similar schema were built:
1. `build-step-data.usecase` + `create-schema.ts` - this combined variables from editor control values + static predefined workflow schema
2. `schema-builder.service` - this did the same + combined previewData (left side of editor) variables, but it redefined all the schemas again etc

The simplification is to always build the entire schema in `build-step-data.usecase` and it now optionally accepts `previewData`, to enrich the schema with these ad-hoc preview variables.

This way it seems a bit easier to track the execution flow and we dont need to maintain parity with duplicated schemas etc.